test/GoogleAuthService.test.js

```diff
@@ -13,3 +14,52 @@ it('_expiresAt returns string of numbers only', () => {
   const res = newService._expiresAt(mockAuthResult)
   expect(res).toEqual(expect.stringMatching(/^[0-9]*$/))
 })
+
+it('grant() returns correct response', async () => {
+  const newService = new AuthService()
+  const gAuth = new GoogleAuth()
+  newService.authInstance = gAuth
+  newService.clientConfig = {
+    scope: 'scope1',
+  }
+  // Test that onResolve callback works.
+  let res = await new Promise((res) => {
+    newService.grant(
+      (googleUser) => res(googleUser),
+      (error) => {
+        console.log(error)
+        expect(false).toBeTruthy()
+      }
+    )
+  })
+  expect(res).toBeInstanceOf(GoogleUser)
+
+  // Test that onReject callback as well as the promise is rejected works.
+  gAuth.currentUser.rejectGrant('access_denied')
+  res = await new Promise((res) => {
+    newService
+      .grant(
+        () => {
+          expect(false).toBeTruthy()
+        },
+        (error) => {
+          expect(error).toMatchObject({
+            error: 'access_denied',
+          })
+          res()
+        }
+      )
+      .catch((error) => {
+        expect(error).toMatchObject({
+          error: 'access_denied',
+        })
+        res()
+      })
+  })
+  expect(res).toBeUndefined()
+
+  // Test that the promise is also returned.
+  gAuth.currentUser.resolveGrant()
+  res = await newService.grant()
+  expect(res).toBeInstanceOf(GoogleUser)
+})
```

Comment:
Yes, I felt the need for some level of testing hence added this. Although this may or may not be the best possible way to write a test at the same time this is what came to my mind after a brainstorming session.

1. Yes adding mock to the name sounds good. Will do
2. Sure, makes sense.